src/prompts/md/migrate_to_hypertables.md

```diff
@@ -5,376 +5,195 @@ description: Comprehensive guide for migrating PostgreSQL tables to TimescaleDB
+-- Check existing primary key/ unique constraints
+SELECT conname, pg_get_constraintdef(oid) as definition
+FROM pg_constraint
+WHERE conrelid = 'your_table_name'::regclass AND contype = 'p' OR contype = 'u';
 ```
 
-**⚠️ Important: Primary Key Rules for Hypertable Migration**
-
-Any existing primary key or unique index **MUST include the partitioning column**.
-
-**Migration strategy:**
+**Rules:** PK/UNIQUE must include partition column
 
-1. **If table has NO primary key:** Don't add one (many time-series use cases don't need PKs)
+**Actions:**
+1. **No PK/UNIQUE:** No changes needed
+2. **PK/UNIQUE includes partition column:** No changes needed
+3. **PK/UNIQUE excludes partition column:** ⚠️ **ASK USER PERMISSION** to modify PK/UNIQUE
 
-2. **If existing PK INCLUDES the partitioning column:** No changes needed
-   - Example: PK(id) and partitioning by 'id' ✓
-   - Example: PK(entity_id, timestamp) and partitioning by 'timestamp' ✓
-
-3. **If existing PK does NOT include partitioning column:** **REQUIRES USER PERMISSION**
-   - Example: PK(id) but partitioning by 'timestamp' ❌
-   
-   **⚠️ ASK USER:** "The existing primary key (id) doesn't include the partitioning column (timestamp). To convert to hypertable, I need to modify the primary key to include both columns: PRIMARY KEY (id, timestamp). This may break existing application code. Is this acceptable?"
-   
-   - **If YES:** Modify PK to include partitioning column
-     ```sql
-     -- DROP CONSTRAINT existing_pk_name;
-     -- ALTER TABLE your_table_name ADD PRIMARY KEY (existing_columns, partition_column);
-     ```
-   
-   - **If NO:** Cannot convert to hypertable. Consider using a different partitioning column that's already in the PK, or removing the PK constraint if business logic allows.
-
-**❌ DO NOT modify primary keys without explicit user permission.**
-
-### Compression Configuration
-
-#### Segment-by Column Selection
+**Example: user prompt if needed:**
+> "Primary key (id) doesn't include partition column (timestamp). Must modify to PRIMARY KEY (id, timestamp) to convert to hypertable. This may break application code. Is this acceptable?"
+> "Unique constraint (id) doesn't include partition column (timestamp). Must modify to UNIQUE (id, timestamp) to convert to hypertable. This may break application code. Is this acceptable?"
 
+If the user accepts, modify the constraint:
 ```sql
--- Analyze cardinality of potential segment columns
-SELECT 
-    'entity_id' as column_name,
-    COUNT(DISTINCT entity_id) as unique_values,
-    COUNT(*) as total_rows,
-    ROUND(COUNT(*)::float / COUNT(DISTINCT entity_id), 2) as avg_rows_per_value
-FROM your_table_name
-UNION ALL
-SELECT 
-    'category',
-    COUNT(DISTINCT category),
-    COUNT(*),
-    ROUND(COUNT(*)::float / COUNT(DISTINCT category), 2)
-FROM your_table_name;
+ALTER TABLE your_table_name DROP CONSTRAINT existing_pk_name;
+ALTER TABLE your_table_name ADD PRIMARY KEY (existing_columns, partition_column);
```

Comment:
Should this either be wrapped in a transaction or as a single `ALTER TABLE` statement?